src/librustdoc/formats/cache.rs

```diff
@@ -359,14 +361,19 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
                 // `public_items` map, so we can skip inserting into the
                 // paths map if there was already an entry present and we're
                 // not a public item.
-                if !self.cache.paths.contains_key(&item.def_id)
-                    || self.cache.access_levels.is_public(item.def_id)
+                if !self.cache.paths.contains_key(&item.def_id.expect_real())
+                    || self.cache.access_levels.is_public(item.def_id.expect_real())
```

Comment:
This isn't the same behavior  - before contains_key would return false for fakes. I have a feeling this will ICE.